parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -473,46 +474,84 @@ loop = do
         stepManyAt = Unison.Codebase.Editor.HandleInput.stepManyAt inputDescription
         stepManyAtM = Unison.Codebase.Editor.HandleInput.stepManyAtM inputDescription
         updateAtM = Unison.Codebase.Editor.HandleInput.updateAtM inputDescription
+
+        addDefaultMetadata
+          :: SlurpComponent v
+          -> Action m (Either Event Input) v ()
+        addDefaultMetadata adds = do
+          let
+            addedVs = Set.toList $ SC.types adds <> SC.terms adds
+            parseResult = (Path.parseHQSplit' . Var.nameStr) <$> addedVs
+            (errs, addedNames) = partitionEithers parseResult
+          case errs of
+            e : _ ->
+              error $ "I couldn't parse a name I just added to the codebase! "
+                      <> e <> "-- Added names: " <> show addedVs
+            _ -> do
+              dm <- resolveDefaultMetadata currentPath'
+              case toList dm of
+                []  -> pure ()
+                dm' -> do
+                  let hqs = traverse InputPatterns.parseHashQualifiedName dm'
+                  case hqs of
+                    Left e -> respond $ ConfiguredMetadataParseError
+                      (Path.absoluteToPath' currentPath')
+                      (show dm')
+                      e
+                    Right defaultMeta -> do
+                      respond DefaultMetadataNotification
+                      manageLinks addedNames defaultMeta Metadata.insert
+
         manageLinks :: [(Path', NameSegment.HQSegment)]
```

Comment:
All I did was pull a big block into a function. I guess a good way to avoid getting called out for lack of comments is to write spaghetti code

Reply:
If you'd prefer to track down the original author (sorry if it was me!) and ask them to commit some comments on this PR, that works for me.  :)  They may have thought they were in the clear since it wasn't flagged on its initial commit, but each time a chunk of code gets fresh eyes on it, it's another chance to improve it / document it.

Also if you don't wanna do either that's okay too, as it's no worse than before, I just hoped you were better positioned to do that since you'd worked with it recently.

Reply:
Only joking, I will definitely add some comments. 👍

Reply:
Thanks! 😅